### DISCUSSION TITLE
How do I get special workspace to slide in from top?

### DISCUSSION THREAD

**Discussion by matthewsia98** [hyprwm/Hyprland#1757] [Q&A]
I have this line in my config `animation = specialWorkspace, 1, $animation_fast_speed, myBezier, slidevert` which makes the special workspace slide in from the bottom.

How do I get it to slide in from the top?

For regular workspaces, I have this line `animation = workspaces, 1, $animation_slow_speed, myBezier, slidevert` and I can see that sliding can be from the top going down or bottom going up depending on whether I am switching to the next or previous workspace. So it seems like the mechanism exists, I just don't see how to achieve this from reading the wiki.

**Comment by cvanelteren** [comment]
The configuration lists that you have to configure it with the keyboard `specialWorkspaces` (https://wiki.hyprland.org/Configuring/Animations/)

In your case it would be

```
animation = specialWorkspaces, 1, $animation_fastSpeed, myBezier, slidevert
```

**Reply by matthewsia98** [reply]
That is exactly what I have

**Reply by cvanelteren** [reply]
You are missing an `s` ;-)!

**Reply by matthewsia98** [reply]
looks like there is no `s` here or are we talking about different things

![image](https://user-images.githubusercontent.com/71731492/235300049-25956362-3650-4646-9882-139ed572578d.png)

**Reply by cvanelteren** [reply]
Ah you are right; weird I have this in my config and it works.

**Reply by cvanelteren** [reply]
My config is as such 
![image](https://user-images.githubusercontent.com/19485143/235300462-705ebe94-444a-4c48-9dc8-20c0b10d4d5b.png)

**Reply by cvanelteren** [reply]
@matthewsia98, apologies I completed read your question wrong. The slide from the top is indeed not supported at the moment. I am also looking to get this work, but I guess it requires some tooling. I will see if I can put a PR; a good place to start would be https://github.com/hyprwm/Hyprland/blob/67c73ec10036c5ff9bda392390ecc7af48d4feee/src/managers/AnimationManager.cpp

**Comment by AQDuck97** [comment]
Hey, 
I would love this feature too! It just feels more natural for workspaces with greater number to appear from the top.
Would love to know if anyone knows of a script or if this is possible now in 2024?

**Comment by ndb000** [comment]
it works if you use slidefadevert instead of slidevert: 
`animation = specialWorkspace, 1, 6, default, slidefadevert -50%` (can be any negative percentage)

**Reply by AQDuck97** [reply]
Thanks! It looks great for special workspace, but for normal workspaces it removes the "chain" effect of seeing the previous workspace getting pushed away since there's a bigger gap and fading.

I'm not going to be too nitpicky though, it does appear from the top and after playing around with speed and offset I got it feeling pretty good.
